quickwit-search/src/root.rs

```diff
@@ -110,21 +113,24 @@ pub struct ErrorRetries {
 /// - Is the node ok, but just the split failing?
 /// - Did all requests fail? (Should we retry in that case?)
 fn analyze_errors(search_result: &SearchResultsByAddr) -> Option<ErrorRetries> {
-    let contains_errors = search_result.values().any(|res| {
-        res.is_err()
-            || res
-                .as_ref()
-                .map(|ok_res| ok_res.failed_requests.is_empty())
-                .unwrap_or(false)
+    let contains_retryable_errors = search_result.values().any(|res| {
```

Comment:
I suspect you can simplify your code by computing `retry_split_ids` explicitly even on your happy path.

Then contains_retryable_errors comes simply as 
```
let contains_retryable_error = retry_split_ids.is_empty();
```

Reply:
That removes some code, so that's better.

Rust doesn't allocate for empty vecs? I didn't heard that before, where can I find something about that? 
Although I wouldn't be concerned about the performance impact here by one vec allocation.